File: 1877. Minimize Maximum Pair Sum in Array/Minimize Maximum Pair Sum in Array.py
class Solution:
    def minPairSum(self, nums: List[int]) -> int:
        nums.sort()
        maxSum = 0
        n = len(nums)
        end = n/2
        i = 0
        while (i < end):
            if (maxSum < (nums[i] + nums[n-1-i])):
                maxSum = nums[i] + nums[n-1-i]
            i = i + 1
        return maxSum

File: 1877. Minimize Maximum Pair Sum in Array/Minimize Maximum Pair Sum in Array.java
class Solution {
    public int minPairSum(int[] nums) {
        Arrays.sort(nums);
        int maxSum = 0;
        int n = nums.length;
        for (int i = 0; i < n/2; i++){
            if (maxSum < (nums[i] + nums[n-1-i])){
                maxSum = nums[i] + nums[n-1-i];
            }
        }
        return maxSum;
    }
}
